pkgs/development/libraries/soil/default.nix

```diff
@@ -1,28 +1,41 @@
-{ stdenv, fetchurl, unzip, mesa, libX11, libGL }:
+{ stdenv, lib
+, Carbon
+, fetchzip
+, libGL
+}:
 
 stdenv.mkDerivation {
   name = "soil";
 
-  src = fetchurl {
-    url    = "http://www.lonesock.net/files/soil.zip";
-    sha256 = "00gpwp9dldzhsdhksjvmbhsd2ialraqbv6v6dpikdmpncj6mnc52";
+  src = fetchzip {
+    url = "https://web.archive.org/web/20200104042737id_/http://www.lonesock.net/files/soil.zip";
+    sha256 = "1c05nwbnfdgwaz8ywn7kg2xrcvrcbpdyhcfkkiiwk69zvil0pbgd";
```

Comment:
Tried downloading myself and got the old `00gpwp9dldzhsdhksjvmbhsd2ialraqbv6v6dpikdmpncj6mnc52` hash, just like I would expect.

Reply:
Yeah I got the same thing for fetchurl too. Fetchzip changes the hash, not exactly sure how that is computed

Reply:
Why switch to fetchzip?

Reply:
Oh, that explains the rehash then. I mostly just like that it conceals the unpacking logic as an implementation detail. Pretty minor though, would be fine to do it manually if that's preferable

Reply:
No preference in general, but I like the idea of keeping either the hash or the URL the same (when possible).

Reply:
Ok, done